provider/infoblox/infoblox.go

```diff
@@ -140,48 +154,44 @@ func NewInfobloxProvider(infobloxConfig InfobloxConfig) (*InfobloxProvider, erro
 // Records gets the current records.
 func (p *InfobloxProvider) Records(ctx context.Context) (endpoints []*endpoint.Endpoint, err error) {
 	zones, err := p.zones()
+
 	if err != nil {
 		return nil, fmt.Errorf("could not fetch zones: %s", err)
 	}
 
 	for _, zone := range zones {
-		logrus.Debugf("fetch records from zone '%s'", zone.Fqdn)
+		endpointsTypeA := make([]*endpoint.Endpoint, 0)
```

Comment:
I think you can initialize this to `len(resA) + len(resH)` or something similar given that you are adding every item just a few lines down.

Reply:
Had been thinking about that, but because of the dynamically adding of items from a-records and h-records (which can contain multiple ip addresses) using an append() is better and prevents mistakes in initializing.